### ISSUE TITLE
How to roll the log by date and size

### ISSUE THREAD

**Issue by doestmatter** [log4js-node/log4js-node#765]
As I know currently log4js does NOT support rolling the log by BOTH date and file size.
Example:
For rolling by date, using dateFile appender.
For rolling by file size, using file appender.

So what if I want to roll by both date & file size?
I did many googled but found nothing.

Thanks in advance!

**Comment by nomiddlename** [comment]
That's something that is in the latest version of the streamroller dependency we use for the log rolling, but I've had trouble getting the latest version to work with log4js. It's on the list for things to do.

**Comment by yqsailor** [comment]
+1

**Comment by nomiddlename** [comment]
This should be fixed in log4js@4.0.0 in npm.

**Comment by GeniusBrother** [comment]
@nomiddlename hi, this is my configuration
```
        access: {
            type: "dateFile",
            pattern: "yyyy-MM-dd-hh",
            keepFileExt: true,
            maxLogSize: 500,
            backups: 3,
            alwaysIncludePattern: true,
            daysToKeep: 15,
            filename: path_1.default.join(appPath_1.default.appDir, "/log/access.log"),
        }
```
but still roll by date

**Comment by nomiddlename** [comment]
Try changing `maxLogSize` to `maxSize` - there's a mismatch between the log4js config (maxLogSize) and the streamroller config (maxSize).

**Comment by GeniusBrother** [comment]
ok, thank you, but the document does not specify this @nomiddlename

**Comment by nomiddlename** [comment]
Yes, you're right - I created another issue to fix that problem #887

**Comment by zhupengfeivip** [comment]
@nomiddlename Hello, why is my code not working as expected?

`
log4js.configure({
    appenders: {
        console: {
            "type": "console",
            "category": "console"
        },
        everything: {
            type: 'dateFile',
            pattern: "yyyy-MM-dd",
            keepFileExt: true,  //
            maxLogSize: 1024 * 1024 * 1, //1024 * 1024 * 1 = 1M
            backups: 2,     //
            alwaysIncludePattern: true,     //
            daysToKeep: 3, //
            filename: 'log/run.log',
        },
    },
    "categories": {
        "default": {
            "appenders": [
                "everything",
                "console"
            ],
            "level": "ALL"
        }
    },
    "pm2": true,
    "replaceConsole": true
});
`

The result of running the above code is
One log file per day, if there are more logs on a day, the maximum number of daysToKeep files
![image](https://user-images.githubusercontent.com/30720809/108812573-542d8b00-75ea-11eb-8400-43ddd192e07a.png)




But here's what I expect
Delete records from daysToKeep days ago. For each log that exceeds the size of the log, click backups to scroll through the record cycle


Please help me
